internal/expr/expr_test.go

```diff
@@ -455,6 +458,39 @@ comment */
 string
 of three lines' AND id = $Person.*`,
 		err: `cannot parse expression: line 3, column 26: asterisk not allowed in input expression "$Person.*"`,
+	}, {
+		query: "SELECT &S[:] FROM t",
+		err:   `cannot parse expression: column 8: cannot use slice syntax "S[:]" in output expression`,
+	}, {
+		query: "SELECT &S[0] FROM t",
+		err:   `cannot parse expression: column 8: cannot use slice syntax in output expression`,
+	}, {
+		query: "SELECT &S[1:5] FROM t",
+		err:   `cannot parse expression: column 8: cannot use slice syntax in output expression`,
+	}, {
+		query: "SELECT col1 AS &S[1:5] FROM t",
+		err:   `cannot parse expression: column 16: cannot use slice syntax in output expression`,
+	}, {
+		query: "SELECT col1 AS &S[] FROM t",
+		err:   `cannot parse expression: column 16: cannot use slice syntax in output expression`,
+	}, {
+		query: "SELECT * FROM t WHERE id IN $ids[:-1]",
+		err:   `cannot parse expression: column 35: invalid slice: expected ']'`,
```

Comment:
You are on fire lately! I like that suggestion. Because we now know there is only one accepted format that would make it clearer.

Reply:
I have changed the error message. The only confusing aspect is with whitespace because we are saying `expected slice[:]` when in reality `slice[ : ]` is also accepted, do you think that is confusing?